platform/broadcom/docker-syncd-brcm/start.sh

```diff
@@ -8,11 +8,22 @@ SYNCD_SOCKET_FILE=/var/run/sswsyncd/sswsyncd.socket
 rm -f ${SYNCD_SOCKET_FILE}
 
 mkdir -p /etc/sai.d/
+rm /etc/sai.d/*
+
+# There are two ways to specify the contents of the SAI_INIT_CONFIG_FILE and they are mutually exclusive
+# via current method (sai.profile.j2) or new method (append.config.j2)
+# If delta is large, use sai.profile.j2 which basically require the user to select which config file to use
+# If delta is small, use append.config.j2 where additional SAI INIT config properties are added
+#   based on specific device metadata requirement
+#   in this case sai.profile should have been modified to use the path /etc/sai.d/sai.config
 
 # Create/Copy the sai.profile to /etc/sai.d/sai.profile
 if [ -f $HWSKU_DIR/sai.profile.j2 ]; then
     sonic-cfggen -d -t $HWSKU_DIR/sai.profile.j2 > /etc/sai.d/sai.profile
 else
+    if [ -f $HWSKU_DIR/append.config.j2 ]; then
```

Comment:
Should this be done irrespective of  `sai.profile.j2 `exists?

Reply:
@prsunny These 2 methods are mutual exclusive.  user must select one of the method (using sai.profile.j2 or  append.config.j2) to manage the config.bcm selection.  
Method 1 via sai.profile.j2 is the existing way and this method requires user to have multiple config.bcm files in its entirely to be selected and used.  no modification on top of this due to the file selected is mounted as Read Only.
Method 2 via append.config.j2 allows the user to use one config.bcm file and add additional logic to selectively include some additional SOC properties within the same config file.  But since this method requires dynamic modification, it must be done at the /etc/sai.d directory which requires upecifying this new path in the final sai.profile file (also under /etc/sai.d/).
So in short. this is purposely done so that it will reduce the complexity and user must select which method to specify the config.bcm file.

Reply:
got it.. thanks @gechiang